qcodes/dataset/measurements.py

```diff
@@ -407,8 +456,13 @@ def register_parameter(
             basis: The parameters that this parameter is inferred from. If
                 this parameter is not inferred from any other parameters,
                 this should be left blank.
+            paramtype: type of the parameter, i.e. the SQL storage class
         """
         # input validation
+        if paramtype not in ParamSpec.allowed_types:
```

Comment:
is this check necessary here, or, instead, can we wait until `ParamSpec` object is initiated? Seems like it's good that it's here because the error message refers to the fact that a qcpdes parameter is being passed in to the method...